src/padplus-manager/padplus-manager.ts

```diff
@@ -591,8 +591,15 @@ export class PadplusManager extends EventEmitter {
             const deleteUserName = contactData.field
             if (this.cacheManager) {
               if (isRoomId(deleteUserName)) {
-                await this.cacheManager.deleteRoomMember(deleteUserName)
-                await this.cacheManager.deleteRoom(deleteUserName)
+                const room = await this.cacheManager.getRoom(deleteUserName)
+                const roomMember = await this.cacheManager.getRoomMember(deleteUserName)
+                if (!room || !roomMember) {
```

Comment:
maybe we should insert a record with only `id` and `delated` flag into the cache here?